common/eqemu_config.cpp

```diff
@@ -111,13 +114,90 @@ void EQEmuConfig::parse_config()
 		DisableConfigChecks = true;
 	}
 
-	/**
-	 * UCS
-	 */
-	ChatHost = _root["server"]["chatserver"].get("host", "eqchat.eqemulator.net").asString();
-	ChatPort = Strings::ToUnsignedInt(_root["server"]["chatserver"].get("port", "7778").asString());
-	MailHost = _root["server"]["mailserver"].get("host", "eqmail.eqemulator.net").asString();
-	MailPort = Strings::ToUnsignedInt(_root["server"]["mailserver"].get("port", "7778").asString());
+	// ucs
+	std::string chat_host = _root["server"]["chatserver"].get("host", "").asString();
+	uint32      chat_port = Strings::ToUnsignedInt(_root["server"]["chatserver"].get("port", "0").asString());
+	std::string mail_host = _root["server"]["mailserver"].get("host", "").asString();
+	uint32      mail_port = Strings::ToUnsignedInt(_root["server"]["mailserver"].get("port", "0").asString());
+
+	// automatic ucs legacy configuration migration
+	// if old configuration values are set, let's backup the existing configuration
+	// and migrate to to use the new fields and write the new config
+	if (!chat_host.empty() || !mail_host.empty()) {
+		LogInfo("Migrating old [eqemu_config] UCS configuration to new configuration");
```

Comment:
Wouldn't it be easier and less to maintain if a deprecation warning is given for a missing `ucs` entry and it just falls back to assigning the ucs server to the `chatserver` or `mailserver` entries? These would always have to be checked anyway with conversion code so they could just continue to exist as aliases for `ucs` without a need to modify configs.

Reply:
It is a good call and it is certainly in the server code, but thinking from a tooling perspective, installer perspective and all supporting things that interact with the configuration - it's easier if we migrate to the new field now rather than checking for all 3 and doing fallbacks in tools to gracefully handle for this as well.

In Spire server configuration editor for example, I'd then have to check their old config value AND if the new one exists and use some small logic to determine what should be shown / saved.